protocols/ping/src/dial_layer.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use futures::prelude::*;
+use libp2p_core::nodes::{ConnectedPoint, NetworkBehavior, NetworkBehaviorAction};
+use libp2p_core::{nodes::protocols_handler::ProtocolsHandler, PeerId};
+use std::marker::PhantomData;
+use tokio_io::{AsyncRead, AsyncWrite};
+use void::Void;
+use PeriodicPingHandler;
+
+/// Network behaviour that handles receiving to pings sent by other nodes.
+pub struct PeriodicPingBehaviour<TSubstream> {
+    /// Marker to pin the generics.
+    marker: PhantomData<TSubstream>,
+}
+
+impl<TSubstream> PeriodicPingBehaviour<TSubstream> {
+    /// Creates a `PeriodicPingBehaviour`.
+    pub fn new() -> Self {
+        PeriodicPingBehaviour {
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<TSubstream> Default for PeriodicPingBehaviour<TSubstream> {
+    #[inline]
+    fn default() -> Self {
+        PeriodicPingBehaviour::new()
+    }
+}
+
+impl<TSubstream> NetworkBehavior for PeriodicPingBehaviour<TSubstream>
```

Comment:
Just noticed that our spelling of behaviour is inconsistent.

Reply:
Yeah, I noticed that and want to fix the typo once everything is merged.